cupy/core/core.pyx

```diff
@@ -1005,6 +1005,8 @@ cdef class ndarray:
             return _math._subtract(x, y)
 
     def __mul__(x, y):
+        if isinstance(y, cupy.poly1d):
+            y = y.coeffs
```

Comment:
1) we cared about devices synchronization issue in polyadd & polysub. But since there is no way to avoid it in polymul, so why not support them? Actually why not supporting them in polyadd & polysub too since we already have room to allow devices synchronization for special cases
2) The tests are more neat in this case for polymul without separating the 16 combinations.

So do you still want to remove it and separate the tests?

Reply:
Supporting only `cupy.ndarray * cupy.poly1d` is confusing for the users.